<?xml version="1.0"?>
<robot name="myrobot">

  <material name="darkGray">
    <color rgba="0.063 0.37 0.3 1" />
  </material>

  <material name="yellow">
    <color rgba="1 0.8 0 1" />
  </material>

  <material name="gray">
    <color rgba="0.4 0.4 0.4 1" />
  </material>

  <link name="neck_link">
    <visual>
      <geometry>
        <mesh filename="package://workshop/meshes/neck_base.stl" />
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0" />
      <material name="darkGray" />
    </visual>
  </link>


  <link name="shoulder_left_1">
    <visual>
      <geometry>
        <mesh filename="package://workshop/meshes/left_shoulder.stl" />
      </geometry>
      <origin rpy="0 -1.570796327 0" xyz="0 0 0" />
      <material name="yellow" />
    </visual>
  </link>

  <link name="shoulder_left_2">
    <visual>
      <geometry>
        <mesh filename="package://workshop/meshes/left_shoulder_2.stl" />
      </geometry>
      <origin rpy="-3.141592653589793 0 -3.141592653589793" xyz="0 0 0" />
      <material name="gray" />
    </visual>
  </link>


  <link name="upper_arm_left">
    <visual>
      <geometry>
        <mesh filename="package://workshop/meshes/upper_arm_left.stl" />
      </geometry>
      <origin rpy="0 1.570796327 0" xyz="0 0 0" />
    </visual>
  </link>


  <link name="mid_arm_left">
    <visual>
      <geometry>
        <mesh filename="package://workshop/meshes/mid_arm.stl" />
      </geometry>
      <origin rpy="0 0 -1.570796327" xyz="0 0 0" />
      <material name="gray" />
    </visual>
  </link>


  <link name="lower_arm_left">
    <visual>
      <geometry>
        <mesh filename="package://workshop/meshes/lower_arm.stl" />
      </geometry>
      <origin rpy="0 0 -1.570796327" xyz="0 0 0" />
    </visual>
  </link>

  <!--############## Right arm links ####################-->

  <link name="shoulder_right_1">
    <visual>
      <geometry>
        <mesh filename="package://workshop/meshes/right_shoulder.stl" />
      </geometry>
      <origin rpy="0 -1.570796327 0" xyz="0 0 0" />
      <material name="yellow" />
    </visual>
  </link>


  <link name="shoulder_right_2">
    <visual>
      <geometry>
        <mesh filename="package://workshop/meshes/right_shoulder_2.stl" />
      </geometry>
      <origin rpy="-3.141592653589793 0 -3.141592653589793" xyz="0 0 0" />
      <material name="gray" />
    </visual>
  </link>


  <link name="upper_arm_right">
    <visual>
      <geometry>
        <mesh filename="package://workshop/meshes/upper_arm_right.stl" />
      </geometry>
      <origin rpy="0 0 -1.570796327" xyz="0 0 0" />
    </visual>
  </link>


  <link name="mid_arm_right">
    <visual>
      <geometry>
        <mesh filename="package://workshop/meshes/mid_arm.stl" />
      </geometry>
      <origin rpy="0 1.570796327 0" xyz="0 0 0" />
      <material name="gray" />
    </visual>
  </link>


  <link name="lower_arm_right">
    <visual>
      <geometry>
        <mesh filename="package://workshop/meshes/lower_arm.stl" />
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0" />
    </visual>
  </link>

  <!--#################### Joints #######################-->


  <joint name="neck_base_shoulder_left" type="revolute">
    <axis xyz="-1 0 0" />
    <limit effort="1000.0" lower="-3.141592653589793" upper="3.141592653589793" velocity="4.0" />
    <origin rpy="3.473205 0 0" xyz="-0.121775 -0.007 -0.0235" />
    <parent link="neck_link" />
    <child link="shoulder_left_1" />
  </joint>

  <joint name="shoulder_shoulder_left" type="revolute">
    <axis xyz="0 0 -1" />
    <limit effort="1000.0" lower="-10.141592653589793" upper="10.141592653589793" velocity="4.0" />
    <origin rpy="0 0 2.373647" xyz="-0.0595 0 0" />
    <parent link="shoulder_left_1" />
    <child link="shoulder_left_2" />
  </joint>


  <joint name="shoulder_upperArm_left" type="revolute">
    <axis xyz="-1 0 0" />
    <limit effort="1000.0" lower="-3.141592653589793" upper="3.141592653589793" velocity="4.0" />
    <origin rpy="3.141592653589793 0 0" xyz="-0.0775 0 -0.0145" />
    <parent link="shoulder_left_2" />
    <child link="upper_arm_left" />
  </joint>

  <joint name="upperArm_midArm_left" type="revolute">
    <axis xyz="0 -1 0" />
    <limit effort="1000.0" lower="-10.141592653589793" upper="0.141592653589793" velocity="4.0" />
    <origin rpy="0 2.6 0" xyz="-0.1985 -0.02025 0" />
    <parent link="upper_arm_left" />
    <child link="mid_arm_left" />
  </joint>

  <joint name="midArm_lowerArm_left" type="revolute">
    <axis xyz="1 0 0" />
    <limit effort="1000.0" lower="-3.141592653589793" upper="3.141592653589793" velocity="4.0" />
    <origin rpy="0 0 0" xyz="-0.083 0.0155 0" />
    <parent link="mid_arm_left" />
    <child link="lower_arm_left" />
  </joint>

  <joint name="neck_base_shoulder_right" type="revolute">
    <axis xyz="1 0 0" />
    <limit effort="1000.0" lower="-3.141592653589793" upper="3.141592653589793" velocity="4.0" />
    <origin rpy="-1.047197551 0 0" xyz="0.121775 -0.007 -0.0235" />
    <parent link="neck_link" />
    <child link="shoulder_right_1" />
  </joint>

  <joint name="shoulder_shoulder_right" type="revolute">
    <axis xyz="0 0 -1" />
    <limit effort="1000.0" lower="-10.141592653589793" upper="10.141592653589793" velocity="4.0" />
    <origin rpy="0 0 -2.094395102" xyz="0.0595 0 0" />
    <parent link="shoulder_right_1" />
    <child link="shoulder_right_2" />
  </joint>

  <joint name="shoulder_upperArm_right" type="revolute">
    <axis xyz="-1 0 0" />
    <limit effort="1000.0" lower="-3.141592653589793" upper="3.141592653589793" velocity="4.0" />
    <origin rpy="1.170796327 0 0" xyz="-0.0775 0 -0.0145" />
    <parent link="shoulder_right_2" />
    <child link="upper_arm_right" />
  </joint>

  <joint name="upperArm_midArm_right" type="revolute">
    <axis xyz="0 0 1" />
    <limit effort="1000.0" lower="-3.141592653589793" upper="3.141592653589793" velocity="4.0" />
    <origin rpy="0 0 2.125" xyz="-0.1985 0 -0.02025" />
    <parent link="upper_arm_right" />
    <child link="mid_arm_right" />
  </joint>

  <joint name="midArm_lowerArm_right" type="revolute">
    <axis xyz="0 1 0" />
    <limit effort="1000.0" lower="-3.141592653589793" upper="3.141592653589793" velocity="4.0" />
    <origin rpy="0 0 0" xyz="0 -0.083 0.0155" />
    <parent link="mid_arm_right" />
    <child link="lower_arm_right" />
  </joint>

</robot>